core/charm/repository/charmhub.go

```diff
@@ -109,7 +109,14 @@ func (c *CharmHubRepository) ResolveForDeploy(arg corecharm.CharmID) (corecharm.
 //  3. In theory we could just return most of this information without the
 //     re-request, but we end up with missing data and potential incorrect
 //     charm downloads later.
-func (c *CharmHubRepository) resolveWithPreferredChannel(charmName string, requestedOrigin corecharm.Origin) (*charm.URL, corecharm.Origin, []corecharm.Platform, transport.RefreshResponse, error) {
+func (c *CharmHubRepository) resolveWithPreferredChannel(charmName string, argOrigin corecharm.Origin) (*charm.URL, corecharm.Origin, []corecharm.Platform, transport.RefreshResponse, error) {
+	c.logger.Tracef("Resolving CharmHub charm %q with origin %v", charmName, argOrigin)
+
+	requestedOrigin, err := c.validateOrigin(argOrigin)
+	if err != nil {
+		return nil, corecharm.Origin{}, nil, transport.RefreshResponse{}, errors.Trace(err)
+	}
+
```

Comment:
Adding the logger trace is good. This bit of code isn't needed in 3.3, it got moved to ResolveWithPreferredChannel().